### PR TITLE
yarn run docs available

### LINKED ISSUE #1338: Incorrect documentation:  `yarn run docs` doesn't work
**Describe the bug**
`README.md` claims that the Typedoc can be formatted using `yarn run docs` but no `docs` script is defined.  This leaves us uncertain how best to generate developer-level documentation.

**To Reproduce**
1. Follow instructions:  enter `yarn run docs`.

**Expected behavior**
A way to format the embedded documentation.

### PR DESCRIPTION
**Pull request by CrisGuzmanS**
Hello @tdonohue 🤚 i hope i solve this issue. i will be waiting for any comment 👍

## References
_Add references/links to any related issues or PRs. These may include:_
* Fixes #1338 

## Description
Developers are able to run `yarn run docs`, then, a folder in ./docs will contain the documentation.

## Instructions for Reviewers

List of changes in this PR:
* typedoc@0.19.2 installed
* docs/* folder added to .gitignore

1. go to project root.
2. run: `yarn run docs`
3.  see the content in `./docs`
image

## Checklist

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [ ] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [ ] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by tdonohue — COMMENTED** [review]
@CrisGuzmanS : Thanks!  Overall this works, but there's a few minor things that I'd recommend cleaning up in this PR before we merge it:

1. We should be using a newer version of typedoc.  This PR looks to install typedoc 0.19.2, but according to the [docs we should be using typedoc 0.22](https://typedoc.org/guides/installation/) (as we are on TypeScript 4.5.5).  So, if you could update to Typedoc 0.22 that'd be best.
2. I'd also recommend a few minor changes to our existing `typedoc.json`, as it doesn't look like it has been updated in a long time.  Here's the minor changes I suggest:
```
  "name": "Angular User Interface for DSpace",
  ...
  "exclude": ["**/*+(spec.ts|index.ts|.json|.json5|.js|.config.ts)", "cypress/**", "webpack/**", "src/themes/**"],
```

I've tested the recommended settings in `typedoc.json`, and verified they simplify the TypeDocs slightly.

Beyond those two small changes, I'm +1 this work.  Thanks.

**Comment by tdonohue** [comment]
@CrisGuzmanS : I think you accidentally misunderstood [my prior reques](https://github.com/DSpace/dspace-angular/pull/2179#pullrequestreview-1382165751)t regarding the `typedoc.json`.  It looks like you removed most of the lines from the `typedoc.json` (and that may be causing some of the issues you are seeing).  My prior request was just to ask you to please modify *two lines*:

1. Change the "name" value to `"Angular User Interface for DSpace"`
2. Change the "exclude value to `["**/*+(spec.ts|index.ts|.json|.json5|.js|.config.ts)", "cypress/**", "webpack/**", "src/themes/**"]`

All other lines in the `typedoc.json` file should remain how they were.  I suspect (but haven't had a chance to test it), that removal of one (or more) of the other configs in your `typedoc.json` file has resulted in the error you are seeing.

So, to summarize: I think we need to keep the upgrade from 0.19 to 0.22. But, only two lines of the `typedoc.json` should be modified.

**Comment by CrisGuzmanS** [comment]
@tdonohue i needed to remove many of the options because they are no longer available from version 0.20: https://github.com/TypeStrong/typedoc/releases/tag/v0.20.0
image

i was looking for another option to replace them, but it looks like there is not another way to replace those options

**Comment by tdonohue** [comment]
@CrisGuzmanS : If you have time to fix the merge conflicts, we can get this reviewed/tested again.  Otherwise, I'll have to move this to the 8.0 board.

**Comment by tdonohue** [comment]
Closing as this hasn't been worked on in over a year and is outdated.  We can always reopen later as needed.
